src/main/scala/org/broadinstitute/dsde/firecloud/service/TrialService.scala

```diff
@@ -259,58 +259,78 @@ final class TrialService
     }
   }
 
-  private def enrollUserInternal(userInfo: UserInfo, status: UserTrialStatus): Future[PerRequestMessage] = {
+  private def enrollUserInternal(userInfo: UserInfo, enabledStatus: UserTrialStatus): Future[PerRequestMessage] = {
 
-    val projectId = status.billingProjectName match {
-      case Some(name: String) => name
+    val statusFuture = enabledStatus.billingProjectName match {
+      case Some(name: String) => Future.successful(enabledStatus)
       case None => {
-        logger.warn(s"User ${userInfo.userEmail} attempted to enroll in trial but no billing project in profile.")
-        throw new FireCloudException("We could not process your enrollment. Please contact support. (Error 56)")
+        Try(claimProjectWithRetries(WorkbenchUserInfo(userInfo.id, userInfo.userEmail))) match {
+          case Success(claimed) =>
+            // persist claimed project name to user's profile
+            val updatedStatus = enabledStatus.copy(billingProjectName = Some(claimed.name.value))
+            thurloeDao.saveTrialStatus(userInfo.id, userInfo, updatedStatus) map {
```

Comment:
Maybe we are not doing this in the most efficient order. Would it make sense to check e.g. `getProjectMembers` _before_ assigning the project to the user, so that we can try with a different project, instead of failing later?

Reply:
This sounds good but is a much larger change than I was hoping to tackle in this PR. Do you feel strongly about this?

Reply:
Out of scope, I just pointed it out so that you might have small opportunities to pull the codebase in that direction while making other changes